### ISSUE TITLE
Support use of IAM Roles

### ISSUE THREAD

**Issue by vladistan** [ind9/gocd-s3-artifacts#40]
Some organizations require use of IAM roles and temp credentials.

It would be nice if this plugin supported it.   Here is more info:

http://docs.aws.amazon.com/java-sdk/latest/developer-guide/prog-services-sts.html

**Comment by ashwanthkumar** [comment]
We have support for IAM Roles through an environment variable called `AWS_USE_IAM_ROLE`. Would help in your use-case? 
Ref - http://oss.indix.com/gocd-s3-artifacts/required_environment_variables.html

**Comment by vladistan** [comment]
Yes I saw that it in the code.   My case is bit different,   I have some nodes running outside of AWS they do have AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY configured, but this is not enough.  

To obtain correct credentials the client first has to create [STSAssumeRoleSessionCredentialsProvider](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/STSAssumeRoleSessionCredentialsProvider.html)  using configured role ARN.   And then pass this new credential provider to the constructor of AWSClient

**Comment by ashwanthkumar** [comment]
Ah sorry, I missed the STS part in the first comment. That's something we don't support today. It should be easy to add though. Would you be interested in taking a stab at it? I guess we can get the `roleArn` and `roleName` through environment variables.

**Comment by vladistan** [comment]
Possibly,  I am super busy these days.  So if anybody else wants to take a stab at this they are more than welcome to do so.  I will try to find some time next month for this  otherwise.
